.github/workflows/lint-test.yml

```diff
@@ -1,14 +1,16 @@
 name: Lint & Test
 
 on:
-  push:
-    branches: [main]
   pull_request:
-    branches: [main]
+  push:
+    branches-ignore:
+      - 'update-snapshots'
 
 jobs:
   lint:
     runs-on: ubuntu-latest
+    # Skip `pull_request` runs on local PRs where `push` is already triggered
```

Comment:
```suggestion
    # Run only on local `push` events or `pull_request` events from forks
```

Reply:
I find my comment makes the actual purpose of the condition clearer (i.e. to skip some runs). I've added a comment next to the `pull_request` trigger to mention that it's to cover PRs created from forked repos.